server/src/main/java/org/elasticsearch/action/search/SearchShardsAction.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.action.ActionType;
+
+public class SearchShardsAction extends ActionType<SearchShardsResponse> {
+    public static final String NAME = "indices:data/read/search_shards";
```

Comment:
one doubt I have around this action name is that the API exposes alias filters info, meaning you can retrieve the filter associated with an alias, which may require some kind of manage privilege that data read does not include? On the other hand do we only return filters for aliases that the current user has privileges to read from? Have you checked with the security team what their thoughts are on this (also in the context of the remote cluster security v2 effort)?